gizmo.py

```diff
@@ -0,0 +1,73 @@
+def hello(name,country="Finland"):
+    print("Hello {0}, how are things in {1}?".format(str(name),str(country)))
+
+def spell(input_str):
+    result=input_str[0]
+    for i in input_str[1:]:
+        result+="."+i
+        
+    return print(result)
+
+def relative_path(subject_identifiers,relpath='./subjects/moc_recording'):
+    names=[]
+    relpath='./subjects/mock_recording'
+    for i in subject_identifiers:
+        res="{0}_{1}.rec".format(relpath,i) #new style
+        #res=relpath+"_"+i+".rec" #old style, not good
+        names.append(res)
+
+    return names
+
+class Gizmo:
+    def __init__(self, name):
+        self.name=name
+
+    def speak(self):
+        print(self.name)
+
+def multiplication_table(zero_out_multiples=None):
+    """ 
+    Summary: 
+    Calculates a multiplication table based on two vectors ranging from 0 to 12 each.
+
+    Extended Summary:
+    The function multiplication_table caclulates a multiplication table for all numbers between 1 to 12. If the input argument zero_out_multiples is not None, multiplies of this input argument in the table will be all set to zero.
+
+    Parameters
+    ----------
+    zero_out_multiplies: 
+        Type: int
+        Description of parameter: The arg is used for the multiplication table. If it is not None, the table returned by the function will have all multiples of the given number set to zero.
+
+    Returns
+    -------
+    res
+        Type: numpy.ndarray
+        Description of parameter: The arg contains the calculated multiplication table. Calculation is based on the outer product.
+    """
+    import numpy as np
+
+    a=np.arange(1,13,1)
+    b=np.arange(1,13,1)
+
+    if zero_out_multiples is not None and isinstance(zero_out_multiples, int):
+        #Version with modulo
+        a[a%zero_out_multiples==0]=0  #okay, I learned something :-)
```

Comment:
yay! :)

Reply:
I admit, I had to google it. I hope this is okay? You see my approach below, probably it would not work for all int between 1-13. I am really not used to working with modulo operator. But I knew from Matlab how to work with index replacement.

Reply:
Googling things is always ok! In this simple exercise, the version you wrote below works just as fine. However, it is useful to know about the modulo operator. It's a useful tool in your toolbox as a programmer. For example, you can use it for "wrapping" behavior. Say we want to normalize a given angle to be between 0-360, so an angle of 369 would become 9. You simply write: `angle % 360` to achieve this.

Reply:
The problem with your implementation is that you zero out multiples in `a` and `b` rather than in the produced table. Say for example we want to zero out multiples of 4. 12 is a multiple of 4. But twelve is also produced when we multiply 2 and 6. This is why test 9 fails.